frost_backup/README.md

```diff
@@ -0,0 +1,123 @@
+# Frost Backup Scheme
+
+Mnemonic encoding for [secp256k1] [Shamir secret shares].
+
+## Overview
+
+A Shamir secret sharing backup scheme for secp256k1 scalars that encodes shares as 25 [BIP39] words. This scheme is intended for use with [FROST] signature implementations so users can back up their individual FROST keys (which are just shamir secret shares). The only thing that distinguishes the scheme slightly from a ordinary Shamir secret sharing backup is the small *[polynomial checksum](#polynomial-checksum)* which is meant to discourage an attacker trying to trick a recovered device into thinking it's part of a different key that it was generated with.
+
+## Format Specification
+
+### Structure
+
+```
+#<share_index> <25 BIP39 words>
+```
+
+#### `share_index`
+
+The share index is a decimal integer prefixed `#` with in the format. The index is never `0`. 
+
+#### Words Bit Layout
+
+The 25 BIP39 words encode 275 bits (11 bits per word):
+
+| Bits     | Length | Words    | Content                |
+|----------|--------|----------|------------------------|
+| 0-255    | 256    | 1-24     | Secret share scalar    |
+| 256-263  | 8      | 24       | [Polynomial checksum](#polynomial-checksum)    |
+| 264-274  | 11     | 25       | [Words checksum](#words-checksum)         |
+
+## Checksums
+
+The checksums are [SHA256] hashes of the input data.
+
+- `share_index` (32 bytes): although share indices are usually small numbers they are encoded in the hash as 32-byte secp256k1 scalar (big endian encoded 256-bit integer).
```

Comment:
This doesnt match the code:

```
    fn compute_words_checksum(
        index: u32,
        scalar: &Scalar<impl Secrecy, impl ZeroChoice>,
        poly_checksum: u16,
    ) -> u16 {
        let hash = Sha256::new()
            .add(index.to_be_bytes())
            .add(scalar)
            .add(poly_checksum.to_be_bytes())
            .finalize();
        // Take first 2 bytes as u16 and shift right to keep only the top 9 bits
...
```

Is it preferable to encode as Scalars for some minor futurproofing towards supporting longer share-indicies (avoiding complexity)?

Reply:
Nice find yeah that's meant to take a Scalar. Looks like I fixed it for the polynomial one but not that one.